tendermint/node.go

```diff
@@ -16,28 +17,25 @@ import (
 )
 
 // NewNode retruns new tendermint node that runs the app.
-func NewNode(logger log.Logger, app abci.Application, root string, seeds string, validator string) (*node.Node, error) {
+func NewNode(logger log.Logger, app abci.Application, root, seeds string, validatorPubKey crypto.PubKey) (*node.Node, error) {
+	if err := os.MkdirAll(filepath.Join(root, "config"), 0755); err != nil {
+		return nil, err
+	}
+	if err := os.MkdirAll(filepath.Join(root, "data"), 0755); err != nil {
+		return nil, err
+	}
 
-	os.MkdirAll(root+"/config", os.FileMode(0755))
 	cfg := config.DefaultConfig()
 	cfg.P2P.Seeds = seeds
 	cfg.SetRoot(root)
 
-	os.MkdirAll(root+"/data", os.FileMode(0755))
-	var me = privval.LoadOrGenFilePV(cfg.PrivValidatorKeyFile(), cfg.PrivValidatorStateFile())
-	nodeKey, err := p2p.LoadOrGenNodeKey(cfg.NodeKeyFile())
-	if err != nil {
-		return nil, err
+	nodeKey := &p2p.NodeKey{
```

Comment:
@krhubert why removing `loadOrGenNodeKey`? The goal of this PR is to use a maximum of stuff already done by Cosmos SDK. We could implement our own system to store the node and validator private keys later on ;)

Reply:
You said that node key will be generated every time engine start, so why we want to load them? Using loadOrGen is misleading for other devs. 

By generating directly pub key it's clear that the key is changing on every start.

Reply:
The goal of this PR is simply to start tendermint with the most similar logic and code than 
cosmos sdk. we can change the behavior later on.

Only the genesis is generated to avoid to add its hardcoded json version to the source code. But it brings some limitation and complicated when we add store to the cosmos app..